### ISSUE TITLE
Support new type of instances

### ISSUE THREAD

**Issue by mogol07** [identiops/terraform-hcloud-k3s#11]
Hi!
According news from 06.06.24 https://docs.hetzner.cloud/whats-new#2024-06 Hetzner deprecated tariffs  cxN1(cx11...cx51). Since 06.09.24 (in one week from now) it would be unavailable from API.

Right now, it's not possible to use it due two problem:

1. Validation checked with regex
`can(regex("^(cp?x[1-5]1|cax[1-4]1|ccx[1-6]3)$", var.node_type))`, 
but it could be easily fixed by
`can(regex("^(cp?x[1-5][1-2]|cax[1-4]1|ccx[1-6]3)$", var.node_type))`
2. But the second problem is more complicated. Network interface name is changed from ens10 to enp7s0. Because other type of instances use older name, it's required lookup variable, and I don't know how to organize it better, beside basic list.
3. BTW, internet gateway has hardcoded type of instance as cx11, it's also need to update, and maybe better to let user to configure it in main.tf?

I also has some other problem, but I am not sure about the reason of it (maybe new Ubuntu 24.04, so I need to test more, before create next ticket ;) )

**Comment by mogol07** [comment]
If replaced hardcoded ens10 to enp7s0, and used only cxN2 instances, then cluster is working on ubuntu 22.04.

**Comment by jceb** [comment]
Yes, the changing instance types are a bit of a challenge. Please submit a PR that updates the regex.

Please also submit a PR for making the gateway type configurable.

Ideally, we'd somehow detect the correct network interface dynamically. Do you want to give it a try?

**Comment by jceb** [comment]
I added support for all new instance types and changed ens10 to enp7s0.
